COHHIO/ryha
# Create dev environment

# List of changes:
- Added files to create a dev container that leverages `docker-compose` to set up a development environment with three services:
    - `app`, the application container that installs the proper R version and uses `renv` to install proper R packages versions. This service also leverages `renv` cache functionality.
    - `db`, which sets up a postgreSQL database
    - `pgadmin`, which sets up pgAdmin (database management system)
- Updated different configuration files (i.e. added new entries to `.gitignore`, updated DESCRIPTION to allow the use of markdown in roxygen comments, updated packages versions in `renv.lock`).
- Improved `create_dm()`. I added a new parameter that reads an environmental variable to determine how the data model is created. This avoids the necessity to change app code to alternate between production and non-production data.
- Added `postgres/populate_dev_database/populate_dev_database.R`, a script to populate dev database.
- Added "Development Environment" section to README.
- Included an example "Production" Dockerfile, which is not used for now (as we deploy to shinyapps.io).

# How to test
You should be able to read the "Development Environment" README section and:
1. Open the repository in a dev container (leveraging `renv` cache)
2. Populate the development database
3. Explore the development database in pgAdmin
4. Connect to the database outside of the dev container (for example, from RStudio in the local machine)

# Additional comments
- Check whether we need to improve the README
- Check whether we need to change filenames
- Check whether the new parameters are documented properly in the roxygen comments

Closes #104
